src/borg/archiver.py

```diff
@@ -652,126 +709,82 @@ def _process(self, *, path, parent_fd=None, name=None,
                 # could trigger an error, e.g. if access is forbidden, see #3209.
                 if not matcher.recurse_dir:
                     return
+                recurse_excluded_dir = True
                 with backup_io('stat'):
                     st = os_stat(path=path, parent_fd=parent_fd, name=name, follow_symlinks=False)
-                recurse_excluded_dir = stat.S_ISDIR(st.st_mode)
-                if not recurse_excluded_dir:
+                if not stat.S_ISDIR(st.st_mode):
                     return
 
             if (st.st_ino, st.st_dev) in skip_inodes:
                 return
             # if restrict_dev is given, we do not want to recurse into a new filesystem,
             # but we WILL save the mountpoint directory (or more precise: the root
             # directory of the mounted filesystem that shadows the mountpoint dir).
-            recurse = restrict_dev is None or st.st_dev == restrict_dev
-            status = None
+            right_dev = restrict_dev is None or st.st_dev == restrict_dev
+
```

Comment:
maybe do not rename variables for now, create a smaller, easier to review diff.

if it is important, this can be done afterwards also.

Reply:
i'll revert